tests/foreman/cli/test_host.py

```diff
@@ -25,66 +37,497 @@ def setUp(self):
+        """@Test: Create a host and then delete it by id.
 
         @Feature: Hosts
 
-        @Assert: Host is created
+        @Assert: Host was deleted
+        """
+        host = entities.Host()
+        host.create_missing()
+        host = Host.create({
+            u'architecture-id': host.architecture.id,
+            u'domain-id': host.domain.id,
+            u'environment-id': host.environment.id,
+            # pylint:disable=no-member
+            u'location-id': host.location.id,
+            u'mac': host.mac,
+            u'medium-id': host.medium.id,
+            u'name': gen_string('alphanumeric'),
+            u'operatingsystem-id': host.operatingsystem.id,
+            # pylint:disable=no-member
+            u'organization-id': host.organization.id,
+            u'partition-table-id': host.ptable.id,
+            u'puppet-proxy-id': self.puppet_proxy['id'],
+            u'root-pass': host.root_pass,
+        })
+        Host.delete({'id': host['id']})
+        with self.assertRaises(CLIReturnCodeError):
+            Host.info({'id': host['id']})
+
+    @tier1
+    def test_positive_delete_by_name(self):
+        """@Test: Create a host and then delete it by name.
+
+        @Feature: Hosts
 
+        @Assert: Host was deleted
         """
-        compute_resource = entities.LibvirtComputeResource(
-            url='qemu+tcp://{0}:16509/system'.format(
-                settings.server.hostname
-            ),
-        ).create()
         host = entities.Host()
         host.create_missing()
-        Host.create({
+        result = Host.create({
             u'architecture-id': host.architecture.id,
-            u'compute-resource-id': compute_resource.id,
             u'domain-id': host.domain.id,
             u'environment-id': host.environment.id,
-            u'interface': 'type=network',
-            u'location-id': host.location.id,  # pylint:disable=no-member
+            # pylint:disable=no-member
+            u'location-id': host.location.id,
+            u'mac': host.mac,
             u'medium-id': host.medium.id,
-            u'name': host.name,
+            u'name': gen_string('alphanumeric'),
             u'operatingsystem-id': host.operatingsystem.id,
-            u'organization-id': host.organization.id,  # pylint:disable=E1101
+            # pylint:disable=no-member
+            u'organization-id': host.organization.id,
             u'partition-table-id': host.ptable.id,
             u'puppet-proxy-id': self.puppet_proxy['id'],
             u'root-pass': host.root_pass,
         })
+        Host.delete({'name': result['name']})
+        with self.assertRaises(CLIReturnCodeError):
+            Host.info({'name': result['name']})
+
+
+class HostUpdateTestCase(CLITestCase):
+    """Tests for updating the hosts."""
+
+    def setUp(self):
+        """Create host to reuse later"""
+        super(HostUpdateTestCase, self).setUp()
+        self.proxies = Proxy.list()
+        assert len(self.proxies) > 0
```

Comment:
Maybe implement something like `self.assertGreater` as it is widely used approach